modules/optimization/examples/bimaterial_mesh/grad.i

```diff
@@ -0,0 +1,103 @@
+[Mesh]
+  [gmg]
+    type = GeneratedMeshGenerator
+    dim = 2
+    nx = 16
+    ny = 16
+    xmin = -4
+    xmax = 4
+    ymin = -4
+    ymax = 4
+  []
+[]
+
+[Variables]
+  [adjoint_T]
+  []
+[]
+
+[Kernels]
+  [conduction]
+    type = MatDiffusion
+    diffusivity = diffusivity
+    variable = adjoint_T
+  []
+[]
+
+[Reporters]
+  [misfit]
+    type = OptimizationData
+  []
+  [data]
+    type = ConstantReporter
+    real_vector_names = 'diffusivity'
+    real_vector_values = '0'
+  []
+[]
+
+[DiracKernels]
+  [pt]
+    type = ReporterPointSource
+    variable = adjoint_T
+    x_coord_name = misfit/measurement_xcoord
+    y_coord_name = misfit/measurement_ycoord
+    z_coord_name = misfit/measurement_zcoord
+    value_name = misfit/misfit_values
+  []
+[]
+
+[BCs]
+  [bottom]
+    type = DirichletBC
+    variable = adjoint_T
+    boundary = bottom
+    value = 0
+  []
+[]
+
+[AuxVariables]
+  [temperature_forward]
+  []
+[]
+
+[Functions]
+  [diffusivity_function]
+    type = ParameterMeshFunction
+    exodus_mesh = parameter_mesh_in.e
+    family = MONOMIAL
+    order = CONSTANT
```

Comment:
check that size of mesh and order and family is consistant with the amount of data coming from the reporter